### ISSUE TITLE
./pipe/docker_engine: Access is denied. In the default daemon configuration on Windows, the docker client must be run elevated to connect

### ISSUE THREAD

**Issue by Kaidanov** [docker/for-win#1679]
### Expected behavior
running docker
### Information
![image](https://user-images.githubusercontent.com/164266/36010914-9c911082-0d5c-11e8-9213-98f349f082fb.png)
Microsoft Windows [Version 10.0.16299.192]
(c) 2017 Microsoft Corporation. All rights reserved.

C:\Users\tzvik>docker ps
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.35/containers/json: open //./pipe/docker_engine: Access is denied. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.

C:\Users\tzvik>docker info
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.35/info: open //./pipe/docker_engine: Access is denied. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.

C:\Users\tzvik>docker version
Client:
 Version:       17.12.0-ce
 API version:   1.35
 Go version:    go1.9.2
 Git commit:    c97c6d6
 Built: Wed Dec 27 20:05:22 2017
 OS/Arch:       windows/amd64
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.35/version: open //./pipe/docker_engine: Access is denied. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.

C:\Users\tzvik>docker version
Client:
 Version:       17.12.0-ce
 API version:   1.35
 Go version:    go1.9.2
 Git commit:    c97c6d6
 Built: Wed Dec 27 20:05:22 2017
 OS/Arch:       windows/amd64
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.35/version: open //./pipe/docker_engine: Access is denied. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.

C:\Users\tzvik>docker --version
Docker version 17.12.0-ce, build c97c6d6

C:\Users\tzvik>docker run hello-world
docker: error during connect: Post http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.35/containers/create: open //./pipe/docker_engine: Access is denied. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.
See 'docker run --help'.

C:\Users\tzvik>docker version
Client:
 Version:       17.12.0-ce
 API version:   1.35
 Go version:    go1.9.2
 Git commit:    c97c6d6
 Built: Wed Dec 27 20:05:22 2017
 OS/Arch:       windows/amd64
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.35/version: open //./pipe/docker_engine: Access is denied. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.

C:\Users\tzvik>docker info
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.35/info: open //./pipe/docker_engine: Access is denied. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.

C:\Users\tzvik>docker --version
Docker version 17.12.0-ce, build c97c6d6

C:\Users\tzvik>docker-compose --version
docker-compose version 1.18.0, build 8dd22a96

C:\Users\tzvik>docker-machine --version
docker-machine version 0.13.0, build 9ba6da9

C:\Users\tzvik>docker run --rm -v c:/Users:/data alpine ls /data
docker: error during connect: Post http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.35/containers/create: open //./pipe/docker_engine: Access is denied. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.
See 'docker run --help'.

C:\Users\tzvik>docker info
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.35/info: open //./pipe/docker_engine: Access is denied. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.

C:\Users\tzvik>docker version
Client:
 Version:       17.12.0-ce
 API version:   1.35
 Go version:    go1.9.2
 Git commit:    c97c6d6
 Built: Wed Dec 27 20:05:22 2017
 OS/Arch:       windows/amd64
error during connect: Get http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.35/version: open //./pipe/docker_engine: Access is denied. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.

C:\Users\tzvik>docker run hello-world
docker: error during connect: Post http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.35/containers/create: open //./pipe/docker_engine: Access is denied. In the default daemon configuration on Windows, the docker client must be run elevated to connect. This error may also indicate that the docker daemon is not running.
See 'docker run --help'.

 
### Steps to reproduce the behavior

  1. Installed 
  2. Run as admin
3.Shared c drive
4.Docker has green light

**Comment by Kaidanov** [comment]
Restarting service didn't help

**Comment by Kaidanov** [comment]
![image](https://user-images.githubusercontent.com/164266/36011072-9ab968bc-0d5d-11e8-842f-27f5fc8fb80a.png)
![image](https://user-images.githubusercontent.com/164266/36011089-b2023b7a-0d5d-11e8-8100-b0e253c5d92d.png)
![image](https://user-images.githubusercontent.com/164266/36011096-bf7be936-0d5d-11e8-93f3-9b9209a660d4.png)
![image](https://user-images.githubusercontent.com/164266/36011108-d08d5282-0d5d-11e8-9f5a-2f3fe94541c9.png)

**Comment by pgayvallet** [comment]
Hello,

`Run as admin` is the issue here. If you start docker as admin, you'll need to be in an admin terminal to be able to use it.

**Comment by Kaidanov** [comment]
OK. Thanks.

**Comment by Kaidanov** [comment]
Is there any smart way to define the daemon?

**Comment by kincki** [comment]
I managed to run docker on windows first verifying that these settings, and running the command at the end:
`>> docker-machine env default ↩️  `

```$Env:DOCKER_TLS_VERIFY = "1" 
$Env:DOCKER_HOST = "tcp://192.168.99.101:2376"
$Env:DOCKER_CERT_PATH = "C:\Users\koray\.docker\machine\machines\default"
$Env:DOCKER_MACHINE_NAME = "default"
$Env:COMPOSE_CONVERT_WINDOWS_PATHS = "true"
# Run this command to configure your shell:
# & "d:\Program Files\Docker Toolbox\docker-machine.exe" env default | Invoke-Expression
```
`>> & "d:\Program Files\Docker Toolbox\docker-machine.exe" env default | Invoke-Expression ↩️ `

**Comment by jacqueskang** [comment]
> Hello,
> 
> `Run as admin` is the issue here. If you start docker as admin, you'll need to be in an admin terminal to be able to use it.

Is there any other way to resolve this? I need to run docker with VSTS build agent which doesn't have admin privileges.

**Comment by JulioIzquierdo** [comment]
> > Hello,
> > `Run as admin` is the issue here. If you start docker as admin, you'll need to be in an admin terminal to be able to use it.
> 
> Is there any other way to resolve this please? I need to run docker with VSTS build agent which doesn't have admin privileges.
> 
> I installed Docker EE in Windows Server 2019 (version 1809) with `Install-Package Docker -ProviderName DockerMsftProvider -Force` and I don't see "docker-users" group being created.
> My outcome of `[System.IO.Directory]::GetAccessControl("\\.\pipe\docker_engine")` is
> 
> ```
> Path Owner                  Access
> ---- -----                  ------
>      BUILTIN\Administrators NT AUTHORITY\SYSTEM Allow  FullControl...
> ```

@jacqueskang , were you able to resolve this?  I'm at the same crossroad.

Thanks! 

-Julio

**Comment by ddebrecenijr** [comment]
> > > Hello,
> > > `Run as admin` is the issue here. If you start docker as admin, you'll need to be in an admin terminal to be able to use it.
> > 
> > 
> > Is there any other way to resolve this please? I need to run docker with VSTS build agent which doesn't have admin privileges.
> > I installed Docker EE in Windows Server 2019 (version 1809) with `Install-Package Docker -ProviderName DockerMsftProvider -Force` and I don't see "docker-users" group being created.
> > My outcome of `[System.IO.Directory]::GetAccessControl("\\.\pipe\docker_engine")` is
> > ```
> > Path Owner                  Access
> > ---- -----                  ------
> >      BUILTIN\Administrators NT AUTHORITY\SYSTEM Allow  FullControl...
> > ```
> 
> @jacqueskang , were you able to resolve this? I'm at the same crossroad.
> 
> Thanks!
> 
> -Julio

Check out https://github.com/tfenster/dockeraccesshelper.

**Comment by KevinBarretodev** [comment]
you need to run The Docker Tool coonsle first and you´ll be able to run files on D
![dockerPro](https://user-images.githubusercontent.com/38572437/57346602-bf0b8a80-7114-11e9-94c2-244b98eccaea.PNG)
ocker or do anything.

**Comment by ericbrumfield** [comment]
Here's a setup I got around in Windows Server 2016 using docker commands in a Jenkins slave/agent on this node.  Instead of running Jenkins under the default Local System account I had to switch it to use a Jenkins user account I created in windows.  For context, the Jenkins agent on this server ran from a scheduled task as this user and the windows service for Jenkins does too.

Docker version:  19.03.5
Windows server based off Windows_Server-2016-English-Full-ECS_Optimized-2020.04.16 AWS image.


Gist of the steps I had to do:

1.  Created a docker and docker-users group in windows.
2.  Granted log on as batch to Jenkins user account for the scheduled task to run at startup.
3.  Added the new Jenkins user to docker and docker-users groups.
4.  Specified "group": "docker" in docker daemon.json.
5.  Make sure to restart the tasks and services involved (docker, jenkins windows service and the scheduled task for the Jenkins agent).
6.  Also setup directory permissions for the Jenkins user regarding Jenkins specific dirs.

End result was being able to work with docker from a non-administrator account in Windows Server 2016.  From all the github threads I've read I'm unsure if docker-users group was necessary, but at least some version of docker in windows uses that for linux based containers that I've read when it's switched to that mode.

I initially tried to use https://github.com/tfenster/dockeraccesshelper , which looks great, but it didn't work for some reason and never changed the acl settings on the pipe.
